### DISCUSSION TITLE
Configuration with 2 Solcast ID's ?

### DISCUSSION THREAD

**Discussion by Captain0Kangaroo** [nathanmarlor/foxess_em#75] [General]
Just setting this up but I have two Solcast sites - One for the Front and one for the Back of the house and therefore two ID's.

What do I do in this situation?

**Comment by Captain0Kangaroo** [comment]
Sorry .... Should have read the other post... Just downloading 1.4.x now to test.

**Reply by nathanmarlor** [reply]
I realised last night that with two Solcast sites you'll likely hit the 50 limit before midnight. Need to invent some clever logic to spread the API calls out throughout the day and depending on the number of sites.

**Reply by Captain0Kangaroo** [reply]
I've also got the Solcast Integration running which will be adding to the API calls. Would it be a good idea to remove this and just use the data in the foxess_em?

**Reply by nathanmarlor** [reply]
Yes, if you're OK with the reduced sensors. Don't want to replace the Solcast integration entirely but happy to provide any sensors deemed important.

**Reply by Captain0Kangaroo** [reply]
Great. I've disabled Solcast to reduce the API calls. TBH I only use today and tomorrow forecast which yours does anyway.

With regard to the solar forecast in the HA energy configuration do you know what should be used for two solar entries/ arrays?

**Reply by nathanmarlor** [reply]
I've only got a single array so not quite sure how it should look. What would be best? What does the Solcast integration do?

**Reply by Captain0Kangaroo** [reply]
Solcast just gives you one sensor which covers both sites however HA expects one sensor for each array.

Wasn't sure if just putting the Solcast sensor (which covered both arrays) in one of my panel setups and leaving the other blank would work.

Currently I'm using Forecast.Solar integration and you can setup multiple arrays and you just add the correct one for your panel setup in the energy configuration:
https://user-images.githubusercontent.com/96144685/201288512-cd1e9ebd-3e75-4f82-a75c-f9d5488ba010.MOV

**Reply by nathanmarlor** [reply]
It should work in the same way - the forecast provided in the energy dashboard is a sum of both Solcast sites so adding it to one array should work out OK.

**Reply by nathanmarlor** [reply]
FYI - just added an update to dynamically calculate refresh intervals based on the number of sites, give it a whirl! :+1:
